src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/InstanceEntryPointTableNode.cs

```diff
@@ -52,9 +56,11 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 {
                     int methodIndex = factory.RuntimeFunctionsTable.GetIndex(method);
 
-                    bool enforceOwningType = false;
-                    ModuleToken moduleToken = factory.SignatureContext.GetModuleTokenForMethod(method.Method.GetTypicalMethodDefinition());
-                    if (moduleToken.Module != factory.SignatureContext.GlobalContext)
+                    // In composite R2R format, always enforce owning type to let us share generic instantiations among modules
```

Comment:
Nit: This check feels like a policy decision we should ask from the ModuleGroup so decisions based on composite-ness are centralized

Reply:
Good point. I have added a new method <code>EnforceOwningType(EcmaModule module)</code> to CompilationModuleGroup.